Cabal/Distribution/Parsec/Types/ParseResult.hs

```diff
@@ -0,0 +1,52 @@
+{-# LANGUAGE CPP #-}
+module Distribution.Parsec.Types.ParseResult where
+
+import Prelude ()
+import Distribution.Compat.Prelude
+
+import Distribution.Parsec.Types.FieldDescr
+import Distribution.Parsec.Types.Common (PError (..), PWarning (..), Position (..))
+import Distribution.Parsec.Class (ParsecBS)
+
+import qualified Data.ByteString as BS
+import qualified Text.Parsec as P
+
+-- | TODO: we could accumulate errors as well
+data ParseResult a = PR [PWarning] (Either PError a)
```

Comment:
One idea would be to use `= PR [PWarning] [PError] (Maybe a)`, so we could accumulate "non-fatal" errors, as field parsing is quite independent.